content/hardware/04.pro/boards/portenta-x8/tutorials/16.edge-ai-docker-container/content.md

```diff
@@ -0,0 +1,1145 @@
+
+This allows the model to run within our system configuration while keeping compatibility with the hardware and software constraints.
+
+As new sensor data is collected, the model classifies flow conditions, providing feedback to the system. If an anomaly is detected, the system can trigger predefined actions, such as:
+
+- Activating an alarm to alert operators of potential system failures.
+- Sending notifications to Arduino Cloud for remote monitoring and logging.
+- Adjusting system parameters dynamically to prevent failures or improve efficiency.
+
+By processing data locally on the Portenta X8, this approach reduces dependence on external servers, reducing response time and providing continuous operation in network-limited environments.
+
+#### Preparing the Docker Deployment for Portenta X8
+
+Instead of pulling the container directly, Edge Impulse provides the necessary components to build a custom Docker container for Portenta X8. These include:
+
+- **Container Image URL:**The base inference container.
+- **Arguments:** API key for authentication and runtime configuration.
+- **Ports to Expose:** Defines how external applications can access the inference server.
+
+This setup allows for custom integration with the Portenta X8’s existing Docker-based environment, providing compatibility with other applications and services running on the device.
+
+#### Selecting Docker Deployment
+
+The deployment process starts by selecting a Docker container as the target environment within Edge Impulse:
+
+![Docker container as deployment method](assets/edge-impulse-deployment-docker1.png)
+
+Edge Impulse provides multiple options, including Jetson based deployments and general-purpose Docker containers. **The standard Docker container with an HTTP interface is used for this application.**
+
+#### Building a Custom Docker Container for Portenta X8
+
+For deployment on the Portenta X8, we use the provided container image, arguments and ports to build a custom container that integrates with the system’s existing services.
+
+![Docker container deployment parameters](assets/edge-impulse-deployment-docker2.png)
+
+The inference workflow consists of the following steps:
+
+- **Sensor Data Acquisition:** The flow rate sensor measures present values.
+- **Feature Extraction:** DSP and spectral analysis are applied to structure data.
+- **Model Prediction:** The trained model classifies the current flow condition.
+- **Response Actions:** If an anomaly is detected, the system takes corresponding action.
+
+By running inference directly on the Portenta X8, this approach can provide latency reduction and autonomous anomaly detection.
+
+#### Running the Model as a Docker Container (Optional)
```

Comment:
Is this really optional or is this the final step of deploying the container?

Reply:
This is an optional step with intention to show the user X8 can run the container with the built model, but I think given notice, it seems very confusing. I'll remove the section and leave he note only for documentation reference.

Reply:
Updated!